src/annotator/anchoring/pdf.js

```diff
@@ -295,21 +305,24 @@ async function anchorByPosition(pageIndex, start, end) {
     page.textLayer.renderingDone
   ) {
     // The page has been rendered. Locate the position in the text layer.
-    const root = page.textLayer.textLayerDiv;
-
-    // Do a sanity check to verify that the page text extracted by `getPageTextContent`
-    // matches the transparent text layer.
     //
-    // See https://github.com/hypothesis/client/issues/3674.
-    if (pageText !== root.textContent) {
-      /* istanbul ignore next */
-      warnOnce(
-        'PDF text layer content does not match page text. This will cause anchoring misalignment.'
```

Comment:
A variation on this warning will be re-introduced in a follow-up PR to handle cases where the page text and text layer content differ beyond just whitespace changes.